lib/output.js

```diff
@@ -887,6 +904,21 @@ function OutputStream(options) {
         self._do_print(output, "throw");
     });
 
+    /* -----[ yield ]----- */
+
+    DEFPRINT(AST_Yield, function(self, output){
+        var star = self.is_star ? "*" : "";
+        output.print("yield" + star);
+        if (self.expression) {
+            output.space();
+            self.expression.print(output);
+        } else if (self.is_star === true) {
```

Comment:
> AST can come from multiple sources.

I guess that's true. I don't know how it's encoded in ESTree format. But the [suggested revision](https://github.com/mishoo/UglifyJS2/pull/1104#discussion_r66504500) to `OPT(AST_Yield)` won't hurt AFAIK.